mcstatus/address.py

```diff
@@ -123,8 +125,18 @@ def resolve_ip(self, lifetime: float | None = None) -> ipaddress.IPv4Address | i
         if self._cached_ip is not None:
             return self._cached_ip
 
+        host = self.host
+        if self.host != "localhost" and sys.platform != "darwin":
```

Comment:
We want to raise this warning and perform the replacement if we **are** on MacOS and the host **is** `localhost`.

For some reason, your if check here was written to instead only run if the host wasn't localhost, and we also weren't on macos. i.e. it would tirgger for any host, except for localhost, if you're not on macos. That's clearly wrong.

This suggestion also applies for this same check in async resolve_ip func.

```suggestion
        if self.host == "localhost" and sys.platform == "darwin":
```

Reply:
oops, yes, I just copied if statement that was before